### PR TITLE
Validate setters against openAPI

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent @pwittrock 

This PR is to validate input setter value against the constraints in openAPI file. These validations happen during set operation for setter values. The PR assumes that the openAPI constraints are handcrafted by the user(see tests for schema). Later PRs will include the way for users to input the schema through cli.

### CODE REVIEW

File: kyaml/setters2/set.go [hunk 1]
```diff
@@ -155,27 +163,54 @@ func (s *Set) substitute(field *yaml.RNode, ext *cliExtension, _ *spec.Schema) (
...
+}
+
+// validateAgainstSchema validates the input setter value against user provided
+// openAI schema
+func validateAgainstSchema(ext *cliExtension, sch *spec.Schema) error {
+	sc := spec.Schema{}
+	sc.Properties = make(map[string]spec.Schema)
+	sc.Properties[ext.Setter.Name] = *sch
+
+	input := map[string]interface{}{}
```
Comment (mortent) [inline kyaml/setters2/set.go:198]: nit: we should be consistent whether we create maps with the `make` function or the literal. Here we both approaches within the same function.

File: kyaml/setters2/set.go [hunk 2]
```diff
@@ -155,27 +163,54 @@ func (s *Set) substitute(field *yaml.RNode, ext *cliExtension, _ *spec.Schema) (
...
+	format := `{"%s" : "%s"}`
+	if yaml.IsValueNonString(ext.Setter.Value) {
+		format = `{"%s" : %s}`
+	}
+
+	inputJSON := fmt.Sprintf(format, ext.Setter.Name, ext.Setter.Value)
+	err := json.Unmarshal([]byte(inputJSON), &input)
```
Comment (mortent) [inline kyaml/setters2/set.go:205]: I might be wrong here, but it seems like this is a lot of code for just creating a map with one key-value pair in it? Do we need to create the json string and then parse it?

File: kyaml/setters2/settersutil/fieldsetter.go [hunk 3]
```diff
@@ -47,6 +50,17 @@ func (fs FieldSetter) Set(openAPIPath, resourcesPath string) (int, error) {
 		Description: fs.Description,
 		SetBy:       fs.SetBy,
 	}
+
+	stat, err := os.Stat(openAPIPath)
```
Comment (mortent) [inline kyaml/setters2/settersutil/fieldsetter.go:54]: Can we add some comments here to explain why we are doing this?

File: kyaml/setters2/set.go (hunk 1 above)
Comment (phanimarupaka) [reply kyaml/setters2/set.go:198]: Agree. Fixed.

File: kyaml/setters2/set.go (hunk 2 above)
Comment (phanimarupaka) [reply kyaml/setters2/set.go:208]: To take advantage of json.Unmarshal. So the problem is, if publisher set required type to be integer and user inputs 4 as value, cli tool takes it as string. We need a way to say that 4 is an integer and not string, similarly true is boolean and not string("true"). And the unmarshalled schema is fed into validate.AgainstSchema. I followed this from their test case. https://github.com/go-openapi/validate/blob/master/doc_test.go#L145

File: kyaml/setters2/settersutil/fieldsetter.go (hunk 3 above)
Comment (phanimarupaka) [reply kyaml/setters2/settersutil/fieldsetter.go:58]: Done.

**Pull request merged**